packages/@atjson/source-gdocs-paste/src/converter.ts

```diff
@@ -173,44 +197,81 @@ GDocsSource.defineConverterTo(OffsetSource, doc => {
       }
     });
 
-  // LineBreaks/Paragraphs may have been created for listItem separators,
-  // so delete those which exist in a list (or immediately after) but not in any list item
-  doc
-    .where(
-      (annotation: Annotation) =>
-        is(annotation, LineBreak) || is(annotation, Paragraph)
-    )
-    .as("lineBreak")
+  // Split lists that have paragraphs not wrapped by any list item
+  convertedLists
     .join(
-      doc.where({ type: "-offset-list" }).as("lists"),
-      (l: Annotation, r: Annotation) => l.start >= r.start && l.end <= r.end + 1
-    )
-    .outerJoin(
-      doc.where({ type: "-offset-list-item" }).as("listItems"),
-      (
-        l: { lineBreak: LineBreak | Paragraph; lists: Array<Annotation<any>> },
-        r: Annotation<any>
-      ) => {
-        return l.lineBreak.start >= r.start && l.lineBreak.end <= r.end;
-      }
+      doc.where({ type: "-offset-paragraph" }).as("paragraphs"),
+      ({ list }, paragraph) =>
+        list.start <= paragraph.start && paragraph.end <= list.end
     )
-    .update(({ lineBreak, listItems }) => {
-      if (listItems.length === 0) {
-        doc.removeAnnotation(lineBreak);
+    .update(({ list, listItems, paragraphs }) => {
```

Comment:
(.________ .)

Could you add a comment about why we need to do this?

Reply:
👍